tool2schema/config.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing import Optional
+
+
+class Config:
+    """
+    Configuration class for tool2schema.
+    """
+
+    def __init__(self, parent: Optional[Config] = None, **settings):
+        self._parent = parent
+        self._settings = settings
+
+    @property
+    def ignore_parameters(self) -> list[str]:
+        """
+        List of parameter names to ignore when creating a schema.
+        """
+        return self._get_setting(Config.ignore_parameters.fget.__name__, ["self", "args", "kwargs"])
+
+    @ignore_parameters.setter
+    def ignore_parameters(self, value: list[str]):
+        self._set_setting(Config.ignore_parameters.fget.__name__, value)
+
+    def reset_default(self):
+        """
+        Reset the configuration to the default settings.
+        """
+        self._settings = {}
+
+    def _get_setting(self, name: str, default):
+        """
+        Get a setting value from the settings dictionary or the parent configuration.
+        If not found, return the default value.
+
+        :param name: Name of the setting
+        :param default: Default value, used when the setting is not found in
+            the settings dictionary and this configuration has no parent
+        :return: The requested setting value
+        """
+        fallback = default if not self._parent else getattr(self._parent, name)
```

Comment:
Should we worry if `name` isn't in `self._parent`? Such that we can return `default`?

Reply:
I don't think so, `getattr` should get the property from the parent which will in turn return either its setting or the default

Reply:
Should this then not be `getattr(self._parent, name, default)`?

Reply:
`getattr` cannot fail to return a value, when the parent is present it will either return the default or ask its parent which will do the same (some parent will return a value eventually)

Reply:
How does `getattr` know its parent when it is a built-in Python function?

Reply:
It doesn't, I'm passing the parent as an argument

Reply:
If the parent is not present getattr will not be invoked

Reply:
Ah yes because `getattr` will call the `@property` method.